packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -189,13 +193,19 @@ export default class Editor implements IEditor {
     ): boolean {
         // Only replace the node when it falls within editor
         if (this.contains(existingNode) && toNode) {
-            this.core.api.transformColor(
-                this.core,
-                transformColorForDarkMode ? toNode : null,
-                true /*includeSelf*/,
-                () => existingNode.parentNode.replaceChild(toNode, existingNode),
-                ColorTransformDirection.LightToDark
-            );
+            const nodeReplaceCallback = () =>
+                existingNode.parentNode?.replaceChild(toNode, existingNode);
+            if (transformColorForDarkMode) {
+                this.core.api.transformColor(
+                    this.core,
+                    toNode,
+                    true /*includeSelf*/,
+                    nodeReplaceCallback,
+                    ColorTransformDirection.LightToDark
+                );
+            } else {
+                nodeReplaceCallback();
+            }

```

Comment:
Does this change the logic? In original code even transformColorForDarkMode is false, we still call transformColor()

Reply:
As far as I can tell if the second parameter is null, transformColor only calls the callback and then returns. Would you like me to update the types in transformColor instead?